barretenberg/cpp/pil/vm2/bc_decomposition.pil

```diff
@@ -0,0 +1,148 @@
+//   38 |               2 |             2 |   0xC | 0xD | *** | ...
+//   39 |               1 |             1 |   0xD | *** | *** | ...
+
+pol commit bytes_to_read;
+pol commit sel_overflow_correction_needed;
+sel_overflow_correction_needed * (1 - sel_overflow_correction_needed) = 0;
+// We need to constrain bytes_to_read = min(WINDOW_SIZE, bytes_remaining). First we do...
+#[BYTECODE_OVERFLOW_CORRECTION_VALUE]
+sel * ((1 - sel_overflow_correction_needed) * (bytes_to_read - WINDOW_SIZE) + sel_overflow_correction_needed * (bytes_to_read - bytes_remaining)) = 0;
+// And then we do sel_overflow_correction_needed = 1 iff bytes_remaining < WINDOW_SIZE.
+// #[LOOKUP_BYTECODE_REMAINING_U16]
+// sel_overflow_correction_needed { bytes_remaining - WINDOW_SIZE } in precomputed.sel_u16 { precomputed.clk }
+
+pol commit bytes_to_read_unary;
+#[LOOKUP_BYTECODE_TO_READ_UNARY]
+sel { bytes_to_read, bytes_to_read_unary } in precomputed.sel_unary { precomputed.clk, precomputed.as_unary };
+
+pol commit sel_pc_plus_0, sel_pc_plus_1, sel_pc_plus_2, sel_pc_plus_3, sel_pc_plus_4,
+           sel_pc_plus_5, sel_pc_plus_6, sel_pc_plus_7, sel_pc_plus_8, sel_pc_plus_9,
+           sel_pc_plus_10, sel_pc_plus_11, sel_pc_plus_12, sel_pc_plus_13, sel_pc_plus_14,
+           sel_pc_plus_15, sel_pc_plus_16, sel_pc_plus_17, sel_pc_plus_18, sel_pc_plus_19,
+           sel_pc_plus_20, sel_pc_plus_21, sel_pc_plus_22, sel_pc_plus_23, sel_pc_plus_24,
+           sel_pc_plus_25, sel_pc_plus_26, sel_pc_plus_27, sel_pc_plus_28, sel_pc_plus_29,
+           sel_pc_plus_30, sel_pc_plus_31, sel_pc_plus_32, sel_pc_plus_33, sel_pc_plus_34,
+           sel_pc_plus_35;
+// TODO: I think we don't need plus_0
+#[BYTECODE_UNARY_RECONSTRUCTION]
+bytes_to_read_unary = sel_pc_plus_0  * 2**0  + sel_pc_plus_1  * 2**1  + sel_pc_plus_2  * 2**2  + sel_pc_plus_3  * 2**3  + sel_pc_plus_4  * 2**4  +
+                      sel_pc_plus_5  * 2**5  + sel_pc_plus_6  * 2**6  + sel_pc_plus_7  * 2**7  + sel_pc_plus_8  * 2**8  + sel_pc_plus_9  * 2**9  +
+                      sel_pc_plus_10 * 2**10 + sel_pc_plus_11 * 2**11 + sel_pc_plus_12 * 2**12 + sel_pc_plus_13 * 2**13 + sel_pc_plus_14 * 2**14 +
+                      sel_pc_plus_15 * 2**15 + sel_pc_plus_16 * 2**16 + sel_pc_plus_17 * 2**17 + sel_pc_plus_18 * 2**18 + sel_pc_plus_19 * 2**19 +
+                      sel_pc_plus_20 * 2**20 + sel_pc_plus_21 * 2**21 + sel_pc_plus_22 * 2**22 + sel_pc_plus_23 * 2**23 + sel_pc_plus_24 * 2**24 +
+                      sel_pc_plus_25 * 2**25 + sel_pc_plus_26 * 2**26 + sel_pc_plus_27 * 2**27 + sel_pc_plus_28 * 2**28 + sel_pc_plus_29 * 2**29 +
+                      sel_pc_plus_30 * 2**30 + sel_pc_plus_31 * 2**31 + sel_pc_plus_32 * 2**32 + sel_pc_plus_33 * 2**33 + sel_pc_plus_34 * 2**34 +
+                      sel_pc_plus_35 * 2**35;
+// Note: the above relation constrains the selectors to be binary when taken together with the lookup into unary.
+
+// Constrain shifted columns.
+bytes_pc_plus_1 = sel_pc_plus_0 * bytes';
```

Comment:
This is the one section i don't fully grok. How come it is not
`bytes_pc_plus_1 = sel_pc_plus_1 * bytes'` - assuming `sel_pc_plus_1` is the unary selector for `bytes_pc_plus_1` and `sel_pc_plus_0` is the selector associated with `bytes`

My thinking
- I agree from your earlier comment that i believe `sel_pc_plus_0` is trivially zero (since we always read the byte).
- From ^, if `bytes_remaining == 1` you would still have something in `bytes_pc_plus_1` (which i dont think we should since it's the overflow byte from the new bytecode).
- Does `sel_pc_plus_35` not get used in this constraint list?

Reply:
You are right! fixed.